Kernel/Process.h

```diff
@@ -159,8 +160,8 @@ class Process
     ProcessID pid() const { return m_pid; }
     SessionID sid() const { return m_sid; }
     bool is_session_leader() const { return m_sid.value() == m_pid.value(); }
-    ProcessGroupID pgid() const { return m_pgid; }
-    bool is_group_leader() const { return m_pgid.value() == m_pid.value(); }
+    ProcessGroupID pgid() const { return m_pg ? m_pg->pgid() : 0; }
```

Comment:
For a future issue: All processes should have a non-zero process group. But that's an existing bug, and this PR doesn't need to fix it.